packages/worker/src/worker.ts

```diff
@@ -924,9 +928,17 @@ export class Worker {
         ),
         { defaultValue: undefined }
       );
-    } finally {
+      // Only shutdown the native worker if we completed without an error.
+      // Otherwise Rust / TS are in an unknown state and shutdown might hang.
+      // A new process must be created in order to instantiate a new Rust Core.
+      // TODO: force shutdown in core?
```

Comment:
If at this point we are potentially in a state where we've diverged from what core thinks is happening (IE: It isn't shut down) - which seems like it could be the case, then yes, I would forcibly just unload core from memory to ensure things are kept consistent.

Alternatively if what you're saying (from the comment above) is process must exit at this point anyway, then I wouldn't bother. I'd just make sure we have a clear error message if someone tries to re-instantiate things that says "nope, just restart the whole process"

Reply:
I think for now let's just ask to restart the process, it'd be simpler.
I put this todo here to think about this.